cpp/JobSystem.hpp

```diff
@@ -0,0 +1,175 @@
+#pragma once
+
+#include "Thread.hpp"
+#include "Functional.hpp"
+#include "Vector.hpp"
+#include "MPMCQueue.hpp"
+
+#include <atomic>
+
+namespace hsd
+{
+    struct Job;
+
+    using JobFn = void(*)(Job);
+    using Counter = std::atomic<size_t>;
+
+    struct Job
+    {
+        JobFn task;
+        void** data = nullptr;
+    };
+
+/*
+    template<u8 numargs, typename T>
+    struct JobData
+    {
+        
+    };
+*/
+
+    enum PRIO
```

Comment:
Wouldn't be an `enum class` better?

Reply:
Probably, but I don't really understand the use case difference of the two, I'm just using this for categorizing.